matchbox-server/src/main/java/ch/ahdis/matchbox/validation/matchspark/OpenAIConnector.java

```diff
@@ -0,0 +1,175 @@
+package ch.ahdis.matchbox.validation.matchspark;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import ch.ahdis.matchbox.CliContext;
+import ch.ahdis.matchbox.engine.cli.MatchboxCli;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import org.springframework.beans.factory.annotation.Value;
+
+/**
+ * The OpenAIConnector class is used to create the http-POST-request to OpenAI.
+ * In order to create the request you need to define a model and an API-key (which is stored as an environment variable).
+ * Four of the used prompts are displayed here as final strings.
+ * Because there were 3 prototypes, a slightly different block of code for the respecting methods. There could be some refactoring done here.
+ */
+public class OpenAIConnector {
+
+    // String names of two OpenAI LLMs used in this project.
+    private static final String MODEL_OLD = "gpt-3.5-turbo";
+    private static final String MODEL = "gpt-4o-mini";
+
+    // The API-key used for the requests. Stored as an environment variable.
+    //private static final String API_KEY = System.getenv("OPENAI_API_KEY");
+    private static String API_KEY;
+
+    // Four of the prompts used in this project.
+    public static final String PROMPT1 = "You are an AI that helps analyze and interpret FHIR resources and their validation results." +
+            "Focus on the issues with severity labeled 'fatal' or 'error' and only give a short summary about those error messages and a plan " +
+            "how to fix those issues.";
```

Comment:
why do we have the PROMPT1 - 4 here? Should not the ones from LLMConnector be used?